lib/toml++/toml.cpp

```diff
@@ -0,0 +1,9 @@
+#if !defined TOML_HEADER_ONLY || TOML_HEADER_ONLY == 1
+#	undef TOML_HEADER_ONLY
+#	define TOML_HEADER_ONLY 0
+#endif
+#ifndef TOML_IMPLEMENTATION
+#define TOML_IMPLEMENTATION
+#endif
```

Comment:
```suggestion

```

This isn't necessary; the lib implementation only has a single file - `toml.cpp` - so disabling `TOML_HEADER_ONLY` doesn't achieve anything.

Disabling `TOML_HEADER_ONLY` only makes a difference when including `toml.h` in multiple translation units.

Reply:
Oh ok, I thought it was mandatory since the docs state

> Enables the library's implementation when TOML_HEADER_ONLY is disabled.
> Not defined by default. Meaningless when TOML_HEADER_ONLY is enabled.

Reply:
That's the documentation for `TOML_IMPLEMENTATION`, not for `TOML_HEADER_ONLY`. Defining `TOML_IMPLEMENTATION` is only required _in one translation unit_ when `TOML_HEADER_ONLY == 0`, and is ignored altogether when `TOML_HEADER_ONLY == 1`.

Since this seems to be a recurring source of confusion:

| `TOML_HEADER_ONLY` | `TOML_IMPLEMENTATION` | What `toml.h` includes: |
|--------------------|-----------------------|-------------------------|
| 0                  | undefined             | Only the 'public' parts |
| 0                  | defined               | Everything              |
| 1                  | undefined*            | Everything              |
| 1                  | defined*              | Everything              |

*`TOML_IMPLEMENTATION` is ignored and does not need to be user-defined at all

Reply:
Also note that since `TOML_HEADER_ONLY == 1` is the default, any precompiled packaged version of the library will still need to have users manually define `TOML_HEADER_ONLY == 0` in their code/build system, otherwise this PR won't achieve anything except linker errors due to ODR violations.

Has that been addressed somewhere I haven't seen? I can't think of a way to solve that without changing `TOML_HEADER_ONLY` to be `0` by default, which is not a change I'm willing to make.

I guess you could programatically alter the default value for `TOML_HEADER_ONLY` during packaging by modifying `preprocessor.h`, but feels way too brittle.

Reply:
Yes, I thought about that. The -DTOML_HEADER_ONLY=0 flags gets added to the pkg-config file, the CMake Config file and to the dependency object, so everyone consuming the compiled dependency will automatically get that flag. A bit like CMake's PUBLIC property

Reply:
Ah, I see. Sounds good to me.